### PR TITLE
Refactor/2818/migrate screenshot button

### PR DESCRIPTION
**Pull request by shaman-apprentice**

### CODE REVIEW

File: visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts [hunk 1]
```diff
@@ -1,95 +1,39 @@
...
+		fixture.componentInstance.makeScreenshotToClipBoard = jest.fn()
 
-			screenshotButtonController.onScreenshotToClipboardEnabledChanged(
-				storeService.getState().appSettings.screenshotToClipboardEnabled
-			)
-
-			expect(screenshotButtonController["_viewModel"].screenshotToClipboardEnabled).toBe(true)
-		})
+		userEvent.click(screen.getByTitle("Copy screenshot to clipboard (Ctrl+Alt+F), export it as a file by (Ctrl+Alt+S)"))
+		expect(fixture.componentInstance.makeScreenshotToClipBoard).toHaveBeenCalled()
```
Comment (knoffi) [inline visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts:28]: This test would pass if we replaced
`
makeScreenShotToClipBoard = ( ) => { }`

Is there a way how we can make this test more "demanding" (=stricter assert part)?

File: visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts [hunk 2]
```diff
@@ -1,95 +1,39 @@
...
-			screenshotButtonController.makeScreenshotToFile()
-			expect(screenshotButtonController["buildScreenShotCanvas"]).toBeCalled()
-		})
+	it("should save to file on click, when screenshot to clipboard is not enabled", async () => {
+		Store.store.dispatch(setScreenshotToClipboardEnabled(false))
+		const { fixture } = await render(ScreenshotButtonComponent, { excludeComponentDeclaration: true })
+		fixture.componentInstance.makeScreenshotToFile = jest.fn()
+
+		userEvent.click(screen.getByTitle("Export screenshot as file (Ctrl+Alt+S), copy it to clipboard by (Ctrl+Alt+F)"))
+		expect(fixture.componentInstance.makeScreenshotToFile).toHaveBeenCalled()
```
Comment (knoffi) [inline visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts:37]: This test would pass if we would replace
`
makeScreenShotToFile = ( ) => { }`

Is there a way how we can make this test more "demanding" (=stricter assert part)?

**Review by knoffi — CHANGES_REQUESTED** [review]
This is not a "hard" request, merely a proposal.

File: visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts (hunk 1 above)
Comment (shaman-apprentice) [reply visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts:28]: I like your way of thinking! 👍 

I applied the boys scout rule here. In an ideal world, the UI shouldn't contain the logic behind it at all, but just use some service, which does all the calculation. Then the UI has only the current test to use the service and the logic should be tested within the service.

Unfortunately the logic was already in the component and it wasn't tested but mocked away at a deeper level. 

So the tests cover the same as before and now we have an Angular component more and an AngularJS component less. Small improvements 🤓

Comment (shaman-apprentice) [reply visualization/app/codeCharta/ui/screenshotButton/screenshotButton.component.spec.ts:28]: I added a reminder to #2410. Maybe we can add a nice e2e test later. But lets keep with small improvements (Angular Migration) for now. Although 👍  for the spirit!

**Pull request merged**